src/main/java/games/strategy/engine/framework/map/download/DownloadFile.java

```diff
@@ -70,20 +66,22 @@ private Thread createDownloadThread(final File fileToDownloadTo) {
         }
         state = DownloadState.DONE;
         try {
-          Files.move(fileToDownloadTo, downloadDescription.getInstallLocation());
+          Files.move(fileToDownloadTo, download.getInstallLocation());
 
           final DownloadFileProperties props = new DownloadFileProperties();
-          props.setFrom(downloadDescription);
-          DownloadFileProperties.saveForZip(downloadDescription.getInstallLocation(), props);
+          props.setFrom(download);
+          DownloadFileProperties.saveForZip(download.getInstallLocation(), props);
 
         } catch (final Exception e) {
           final String msg = "Failed to move downloaded file (" + fileToDownloadTo.getAbsolutePath() + ") to: "
-              + downloadDescription.getInstallLocation().getAbsolutePath();
+              + download.getInstallLocation().getAbsolutePath();
           ClientLogger.logError(msg, e);
         }
       }
-      // notify listeners we finished the download
-      downloadCompletedListeners.forEach(e -> e.run());
+
+      watcher.stop();
+      // TODO: update code in this method so we pass reason (DONE, CANCELLED) to the listener
```

Comment:
I don't know if you're planning a follow-up PR to add cancellation support.  If not anytime soon, I'd say go ahead and remove this TODO.